### PR TITLE
BaSP-TG-92: Readjustment of the routes

### PR DESCRIPTION
**Pull request by nicomicheletti**
- Create PrivateRoute
- Create component to show as fallback in case a route has no component yet or the route does not exists
- Add template routes for each entity
- Modularize routes based on entities

Note: all routes are working except for the redirecting route on the Log in page. There's an error on the login component that should be fixed before testing the redirect (out of the scope of this PR)

### CODE REVIEW

File: src/routes/auth.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Switch, Route, Redirect, useRouteMatch } from 'react-router-dom';
+import Login from 'Components/Auth/Login';
+import Layout from 'Components/Layout';
+import register from 'Components/Auth/RegisterForm';
```
Comment (CorinaArreguez) [inline src/routes/auth.jsx:4]: register tiene que ir con mayúscula: Register.

File: src/routes/index.jsx [hunk 2]
```diff
@@ -1,61 +1,23 @@
...
-          </Switch>
-        </Suspense>
-      </Layout>
+      <Suspense fallback={<div />}>
+        <Switch>
+          <Route path="/employee" component={EmployeeRoutes} />
+          <Route path="/admin" component={AdminRoutes} />
+          <Route path="/superadmins" component={superAdminRoutes} />
+          <Route path="/auth" component={AuthRoutes} />
+          <Redirect to="/auth" />
```
Comment (lucasrendo) [inline src/routes/index.jsx:18]: paths are not exact and the redirect does not redirect anywhere as it has the same path as the route component

File: src/routes/superAdmins.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Switch, Route } from 'react-router-dom';
+import Layout from 'Components/Layout';
+
+const superAdmin = () => {
+  return (
+    <Layout>
+      <Switch>
+        <Route path="/superadmins" component={superAdmin} />
```
Comment (lucasrendo) [inline src/routes/superAdmins.jsx:9]: route should be exact and other routes should be added to establish conventions (other routes can be commented)

**Review by lucasrendo — CHANGES_REQUESTED** [review]
- In the register file there is a '/login' route that should be updated to '/auth/login'
- If i use an incorrect employee i get an empty modal while loading and then a whitescreen from which i cannot come back without reloading the page
- The page home should not be the login. It should be the home.

**Review by lucasrendo — APPROVED** [review]

File: src/routes/superAdmins.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  return (
+    <Switch>
+      <Route exact path="/superadmin" component={List} />
+      <Route exact path="/superadmin/add-admin" component={Form} />
+      <Route component={NotFound} />
+    </Switch>
+  );
+};
+
+export default superAdmin;
```
Comment (GonzaloBracamonte) [inline src/routes/superAdmins.jsx:17]: Creo que deberia ser SuperAdmin para mantener una consistencia con los demas controladores de rutas

**Review by GonzaloBracamonte — APPROVED** [review]
Looks good, nice job!

**Pull request merged**